x/profiles/legacy/v8/store.go

```diff
@@ -0,0 +1,131 @@
+package v6
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+
+	"github.com/desmos-labs/desmos/v4/x/profiles/types"
+)
+
+// MigrateStore performs in-place store migrations from v8 to v9.
+// The migration includes:
+//
+// - fixing the chain links so that their types are correct
+func MigrateStore(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec, amino *codec.LegacyAmino) error {
+	store := ctx.KVStore(storeKey)
+
+	// Migrate all the chain links
+	err := migrateChainLinks(store, cdc, amino)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// migrateChainLinks migrates the chain links from v5 to v7 by changing the various Protobuf interface types.
+// The migration from v5 to v6 is skipped because the two types are identical (from v5 to v6 no changes were made).
+func migrateChainLinks(store sdk.KVStore, cdc codec.BinaryCodec, amino *codec.LegacyAmino) error {
```

Comment:
This was added because, as detailed [here](https://github.com/desmos-labs/desmos/pull/1003#issuecomment-1236637358) by @dadamu, during the `v4.3.0` migration the chain links weren't migrated properly. This ended up setting all chain links' signature value type to `SIGNATURE_VALUE_TYPE_COSMOS_DIRECT` while it should have been different based on the plain text value. 

This migration script solves that problem by iterating over the various chain links, checking their plain text value and updating the signature vale type accordingly to the plain text format.